### PR TITLE
scripthaus: init at 0.5.1

### PR DESCRIPTION
**Pull request by raspher**
## Description of changes
needed by #280566

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by j-hui** [comment]
Result of `nixpkgs-review pr 286194` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scripthaus</li>
  </ul>
</details>

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! One thing I would say is this is pretty different from how go packages are normally written, most of the time you should not need to specify the phases explicitly, nor is that usually the fetcher we'd use for GH.

I would suggest using something like `nix-init` to write a more conformant package.

**Comment by raspher** [comment]
@eclairevoyant i've tried the 'normal' way and got stuck at tests and some compilation errors. Then i messed up the code... Now after re-reading the manual i've set `CGO_ENABLED = 1;` as it shoud be done and (sadly) disabled tests.

Thanks for review and for mentioning `nix-init`, what a great tool i didn't know about!

**Comment by raspher** [comment]
upstream fixed tests and released new version, version bumped and enabled tests

File: pkgs/by-name/sc/scripthaus/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  vendorHash = "sha256-GUZNPLBgqN1zBzCcPl7TB9/4/Yk4e7K6I20nVaM6ank=";
+
+  CGO_ENABLED = 1;
+
+  ldflags = [ "-s" "-w" ];
+
+  postInstall = ''
+    mv $out/bin/cmd $out/bin/${pname}
```
Comment (eclairevoyant) [inline pkgs/by-name/sc/scripthaus/package.nix:24]:
```suggestion
    mv $out/bin/cmd $out/bin/scripthaus
```
to match `meta.mainProgram`

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sc/scripthaus/package.nix (hunk 1 above)
Comment (raspher) [reply pkgs/by-name/sc/scripthaus/package.nix:24]: i'm not golang expert (even golang programmer, i'm c# 🙂). On their [github](https://github.com/scripthaus-dev/scripthaus) they say it should be build using `CGO_ENABLED=1 go build -o scripthaus cmd/main.go`. I've tried other ways but without success, so i thought "why not KISS?". 

Do you have any idea to improve this?

**Review by eclairevoyant — APPROVED** [review]
Builds and runs on x86_64-linux. Simple example from the upstream README works as well.

**Comment by raspher** [comment]
added updateScript and tested it locally with `nix-shell maintainers/scripts/update.nix --argstr maintainer raspher`

**Pull request merged**